plugins/utils/competitest.nix

```diff
@@ -0,0 +1,257 @@
+        to the preference specified in testcases_use_single_file
+      '';
+
+      testcases_single_file_format = helpers.defaultNullOpts.mkStr "$(FNOEXT).testcases" ''
+        String representing how single testcases files should be named
+        (see [file-format modifiers](https://github.com/xeluxee/competitest.nvim?tab=readme-ov-file#file-format-modifiers))
+      '';
+
+      testcases_input_file_format = helpers.defaultNullOpts.mkStr "$(FNOEXT)_input$(TCNUM).txt" ''
+        String representing how testcases input files should be named
+        (see [file-format modifiers](https://github.com/xeluxee/competitest.nvim?tab=readme-ov-file#file-format-modifiers))
+      '';
+
+      testcases_output_file_format = helpers.defaultNullOpts.mkStr "$(FNOEXT)_output$(TCNUM).txt" ''
+        String representing how testcases output files should be named
+        (see [file-format modifiers](https://github.com/xeluxee/competitest.nvim?tab=readme-ov-file#file-format-modifiers))
+      '';
+
+      companion_port = helpers.defaultNullOpts.mkInt 27121 ''
+        Competitive companion port number
+      '';
+
+      receive_print_message = helpers.defaultNullOpts.mkBool true ''
+        If true notify user that plugin is ready to receive testcases, problems and
+        contests or that they have just been received
+      '';
+
+      template_file =
+        helpers.mkNullOrOption
+        (
+          with types;
+            oneOf [
+              (enum [false])
+              str
+              (attrsOf str)
+            ]
+        )
+        ''
+          Templates to use when creating source files for received problems or contests.
+          Can be one of the following:
+          * false: do not use templates
+          * string with
+            [file-format modifiers](https://github.com/xeluxee/competitest.nvim?tab=readme-ov-file#file-format-modifiers):
+            useful when templates for different file types have a regular file naming
+          * table with paths: table associating file extension to template file
+        '';
+
+      evaluate_template_modifiers = helpers.defaultNullOpts.mkBool false ''
+        Whether to evaluate
+        [receive modifiers](https://github.com/xeluxee/competitest.nvim?tab=readme-ov-file#receive-modifiers)
+        inside a template file or not
+      '';
+
+      date_format = helpers.defaultNullOpts.mkStr "%c" ''
+        String used to format `$(DATE)` modifier (see
+        [receive modifiers](https://github.com/xeluxee/competitest.nvim?tab=readme-ov-file#receive-modifiers)).
+        The string should follow the formatting rules as per Lua's
+        `[os.date](https://www.lua.org/pil/22.1.html)` function.
+      '';
+
+      received_files_extension = helpers.defaultNullOpts.mkStr "cpp" ''
+        Default file extension for received problems
+      '';
+
+      received_problems_path =
+        helpers.defaultNullOpts.mkNullable (with types; either str helpers.nixvimTypes.rawLua)
```

Comment:
```suggestion
        helpers.defaultNullOpts.mkStr
```
`mkStr` would be sufficient here. Indeed, as most of the other helpers under `helpers.defaultNullOpts`, it allows for `rawType` by default.
Notice the `maybeRaw` [here](https://github.com/nix-community/nixvim/blob/f2f97d844bb39559f3356e209b49c92900d860b8/lib/options.nix#L138).